Fix translation between uri and local file path on Windows (#94)

deathaxe opened the pull request.

deathaxe committed:
Fix translation between uri and local file path on Windows (#94)

palantirtech commented:
Thanks for your interest in [palantir/python-language-server](https://github.com/palantir/python-language-server), @deathaxe! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

deathaxe committed:
Optimize path translation from local to uri and add some tests

see: https://stackoverflow.com/questions/11687478/convert-a-filename-to-a-file-url

deathaxe committed:
Use path translation in language_server

deathaxe committed:
Fix issues in config.find_parents (#96)

The function suffers from 3 issues:

1. If path contains a directory, the function does not search for config files in path
   but starts with the parent directory. Config files located in the same directory as
   the document being handled, are ignored.

   Therefore os.path.dirname() is initially called for files only.

2. If no config file was found in one of the directories the function returns None
   while returning an empty list by the initial checks.

   The function is changed to always return empty lists.

3. Windows like path and root arguments cause the function to never stop on Windows systems.

   Therefore the loop is changed to stop if os.path.dirname() returns its argument
   indicating the root is reached rather than handle os specific path structures.
   This way the function is OS independent now.

gatesn commented:
@deathaxe I setup an AppVeyor build and fixed the tests to run on Windows in #99 

Hopefully this fixes a lot of the Windows path bugs. I think the recursive config bug still exists however. Would you mind starting a new PR if you want to take a look at that? Also let me know if you run into any path problems on Windows as well.

gatesn commented:
Closed in favour of #101 and #99
